packages/components/RadioGroup/src/RadioButton.tsx

```diff
@@ -9,21 +9,27 @@ import { filterViewProps } from '@fluentui-react-native/adapters';
 import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
 import { settings, radioButtonSelectActionLabel } from './RadioButton.settings';
 import { mergeSettings } from '@uifabricshared/foundation-settings';
-import { foregroundColorTokens, textTokens, borderTokens, backgroundColorTokens } from '@fluentui-react-native/tokens';
-import { useAsPressable } from '@fluentui-react-native/interactive-hooks';
+import { foregroundColorTokens, textTokens, borderTokens, backgroundColorTokens, getPaletteFromTheme} from '@fluentui-react-native/tokens';
+import { useAsPressable, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
 import { RadioGroupContext } from './RadioGroup';
 
 export const RadioButton = compose<IRadioButtonType>({
   displayName: radioButtonName,
 
   usePrepareProps: (userProps: IRadioButtonProps, useStyling: IUseComposeStyling<IRadioButtonType>) => {
-    const { content, buttonKey, disabled, ariaLabel, ...rest } = userProps;
+    const { content, buttonKey, disabled, ariaLabel, componentRef, ...rest } = userProps;
+
+    // Since we're overriding pressable's onFocus below, pressable.state.focused does not get updated.
+    // We need to track whether the RadioButton is focused or not manually.
+    const [isButtonFocused, setIsFocused] = React.useState(false);
```

Comment:
Did you consider using our existing useFocusState hook? I was wondering if you had a particular reason for doing it this way

Reply:
I was thinking since I already had to create a custom function for the root's onFocus(), I might as well just do this. And to be honest, I wasn't aware of the useFocusState hook. I just tried to implement it, but it broke it (every radiobutton had the focus rect around it). I think overriding the onFocus() is causing problems.

Would it be sufficient to leave as is?

Reply:
So the reason this won't work is because I'm already setting an _onFocus_ method on the root. The useFocusState's onFocus will be overridden by this one, and thus the focus state won't get updated. That's why I had to modify my already-created onFocus method.

Reply:
You wouldn't be able to leverage onStateChange with useFocusState?

Reply:
The pressable's focus state never changes because I'm overriding the onFocus method that would have been used for the pressable state change. But maybe I'm misunderstanding how onStateChange works? I'm assuming it's just a callback when there's a state change.

Reply:
So the problem I'm having is this:

I feed my onFocus function as a prop to useFocusState. However, my onFocus function references the state of the RadioButton, so I can't feed it as a prop to useFocusState because that gets defined before the state gets defined (because state relies on that).

Maybe there's an easy solution that I'm not seeing at the moment?

Reply:
I think you can implement what you want to do with this new state definition.

Instead of defining this new isButtonFocused state, you can just call pressable.props.onFocus() inside your onFocusChange callback. Pressable.props' onFocus and onBlur should be managing the focus state already so you wouldn't need to reassign basically same state callbacks.

Reply:
Thanks you, updated in newest commit